tri_mocap_common/src/mocap_tf_publisher_node_main.ros2.cpp

```diff
@@ -0,0 +1,15 @@
+#include <memory>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include "mocap_tf_publisher_node.ros2.hpp"
+
+int main(int argc, char** argv)
+{
+  auto args = rclcpp::init_and_remove_ros_arguments(argc, argv);
```

Comment:
It looks like removing ROS arguments from the command line would prevent passing command line arguments to remap topic names or node names ([exampe: `--ros-args --remap mocap_state:=foobar`](https://design.ros2.org/articles/ros_command_line_arguments.html)). Is that intentional? Allowing remapping would make the `tf_topic` and `tracker_topic` parameters unnecessary.

Reply:
I'm not really sure why @hidmic used this structure elsewhere in the repo. Is there documentation on which arguments get consumed by `rcl::init*` versus `NodeOptions`? I'm assuming from the `rclcpp` API that doing
```
rclcpp::init(argc, argv);
rclcpp::spin(make_shared<MyNode>(rclcpp::NodeOptions()));
```
does the right thing about making the full arguments available?

Reply:
> I'm assuming from the rclcpp API that doing [...] does the right thing about making the full arguments available?

Yes, that will do the right thing.

> Is there documentation on which arguments get consumed by `rcl::init*` versus `NodeOptions`?

Neither consumes the arguments. There's some explanation [on the documentation of the remap functions](https://github.com/ros2/rcl/blob/191b13bc2eb8578e40799eb6b50c74b885e7066e/rcl/include/rcl/remap.h#L52-L107). The arguments given to `rclcpp::init` are "global arguments", and ones given directly to a node are "local arguments". Local arguments take precedence, and [nodes can be prevented from using the global arguments completely](https://github.com/ros2/rclcpp/blob/b953bdddf8de213b4a051ecf2d668dad65ff9f89/rclcpp/include/rclcpp/node_options.hpp#L150-L160). The difference between global and local only matters when composing multiple nodes into a process.

Sort-of exceptions are the `enclave` and log configuration arguments. The `enclave` argument is only used by `rclcpp::init()` and applies to the whole context. The logging configuration arguments apply to the whole process. They still can be passed to the node, but they are only used when passed to `rclcpp::init()`.

Reply:
I've redone all of the node entry points to follow the
```
rclcpp::init(argc, argv);
rclcpp::spin(make_shared<MyNode>(NodeOptions()));
```
structure.